input/postgres/relation_stats_ext.go

```diff
@@ -0,0 +1,96 @@
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/lib/pq"
+	"github.com/pganalyze/collector/state"
+	"github.com/pganalyze/collector/util"
+)
+
+const extendedStatisticsSQLExprsField = "null"
+const extendedStatisticsSQLpg14ExprsField = "pg_get_statisticsobjdef_expressions(s.oid)::text[]"
+const extendedStatisticsSQLInheritedField = "null"
+const extendedStatisticsSQLpg15InheritedField = "sd.inherited"
+
+const extendedStatisticsSQL = `
+SELECT c.oid,
+	   n.nspname,
+	   s.stxname,
+	   (SELECT array_agg(k) FROM unnest(s.stxkeys) k) stxkeys,
+	   COALESCE(%s, ARRAY[]::text[]) exprs,
+	   s.stxkind,
+	   %s,
+	   sd.n_distinct,
+	   sd.dependencies
+  FROM pg_catalog.pg_statistic_ext s
+  JOIN pg_class c ON (s.stxrelid = c.oid)
+  JOIN pg_namespace n ON (s.stxnamespace = n.oid)
+  LEFT JOIN %s sd ON (sd.statistics_schemaname = n.nspname AND sd.statistics_name = s.stxname)
+ WHERE ($1 = '' OR (n.nspname || '.' || c.relname) !~* $1)
+`
+
+func GetRelationStatsExtended(ctx context.Context, logger *util.Logger, db *sql.DB, postgresVersion state.PostgresVersion, server *state.Server, globalCollectionOpts state.CollectionOpts, systemType string, dbName string) (state.PostgresRelationStatsExtendedMap, error) {
+	var sourceTable string
+	var exprsField string
+	var inheritedField string
+
+	if postgresVersion.Numeric >= state.PostgresVersion14 {
+		exprsField = extendedStatisticsSQLpg14ExprsField
+	} else {
+		exprsField = extendedStatisticsSQLExprsField
+	}
+
+	if postgresVersion.Numeric >= state.PostgresVersion15 {
+		inheritedField = extendedStatisticsSQLpg15InheritedField
+	} else {
+		inheritedField = extendedStatisticsSQLInheritedField
+	}
+
+	if StatsHelperExists(ctx, db, "get_relation_stats_ext") {
+		logger.PrintVerbose("Found pganalyze.get_relation_stats_ext() stats helper")
+		sourceTable = "pganalyze.get_relation_stats_ext()"
+	} else {
+		if systemType != "heroku" && !connectedAsSuperUser(ctx, db, systemType) && globalCollectionOpts.TestRun {
```

Comment:
I was wondering if we should still warn if you don't actually _have_ any extended statistics, but I think we should, since they often come in handy sooner or later, and it's a good opportunity to have this set up sooner rather than later.